srcpkgs/python3-GitPython/template

```diff
@@ -1,18 +1,39 @@
 # Template file for 'python3-GitPython'
 pkgname=python3-GitPython
-version=2.1.11
-revision=5
+version=3.1.26
+revision=1
 wrksrc="GitPython-${version}"
 build_style=python3-module
+make_check_args="--deselect test/test_base.py::TestBase::test_with_rw_remote_and_rw_repo
+ --deselect test/test_git.py::TestGit::test_refresh
+ --deselect test/test_remote.py::TestRemote::test_base
+ --deselect test/test_diff.py::TestDiff::test_diff_with_staged_file
+ --deselect test/test_docs.py::Tutorials::test_init_repo_object
+ --deselect test/test_docs.py::Tutorials::test_references_and_objects
+ --deselect test/test_refs.py::TestRefs::test_head_reset
+ --deselect test/test_repo.py::TestRepo::test_rebasing
+ --deselect test/test_submodule.py::TestSubmodule::test_git_submodules_and_add_sm_with_new_commit"
 hostmakedepends="python3-setuptools"
 depends="python3-gitdb"
+checkdepends="${depends} python3-pytest-sugar python3-coverage python3-pytest-cov python3-toml
+ python3-mypy flake8 python3-ddt git python3-virtualenv"
 short_desc="Library to interact with Git repositories (Python3)"
-maintainer="Orphaned <orphan@voidlinux.org>"
+maintainer="Jan Christian Grünhage <jan.christian@gruenhage.xyz>"
 license="BSD-3-Clause"
 homepage="https://github.com/gitpython-developers/GitPython"
-distfiles="${PYPI_SITE}/G/GitPython/GitPython-${version}.tar.gz"
-checksum=8237dc5bfd6f1366abeee5624111b9d6879393d84745a507de0fda86043b65a8
+distfiles="https://github.com/gitpython-developers/GitPython/archive/refs/tags/${version}.tar.gz"
+checksum=47c82a6370a186bd426be2e8435ddbb60b254fb86f3fad2ed4028e6237e9d3d1
 
 post_install() {
 	vlicense LICENSE
 }
+
+pre_check() {
+	pushd .
+	cd /tmp
+	git clone https://github.com/gitpython-developers/GitPython
+	cd GitPython
+	yes | ./init-tests-after-clone.sh
+	export GIT_PYTHON_TEST_GIT_REPO_BASE="$(pwd)"
+	popd
+}
```

Comment:
Can we git clone in $wrksrc?

Reply:
You mean instead of pulling in and extracting the distfile? Should be possible, yeah

Reply:
I meant just git cloning gitpython to a separate directory inside $wrksrc, so we don't pollute /tmp and get it properly removed when do_check() fails.

Reply:
right, done that instead now.